Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -3294,20 +3312,7 @@ private void getPlayableFromGraveyardCard(Game game, Card card, Abilities<Abilit
             if (fromAll || fromZone == Zone.EXILED) {
                 for (ExileZone exile : game.getExile().getExileZones()) {
                     for (Card card : exile.getCards(game)) {
-                        if (null != game.getContinuousEffects().asThough(card.getId(),
-                                AsThoughEffectType.PLAY_FROM_NOT_OWN_HAND_ZONE, null, this.getId(), game)) {
-                            for (Ability ability : card.getAbilities()) {
-                                if (ability.getZone().match(Zone.HAND)) {
-                                    ability.setControllerId(this.getId()); // controller must be set for case owner != caster
-                                    if (ability instanceof ActivatedAbility) {
-                                        if (((ActivatedAbility) ability).canActivate(playerId, game).canActivate()) {
```

Comment:
You are missing from `canActivate` call on code refactor to `cardPlayableAbilities`

Reply:
Oh, I see. Is there a reason `canActivate` isn't there for the graveyard + revealed cards zones?

Reply:
It's interesting... Maybe another bug/issue:
* Graveyard uses it in getPlayableFromGraveyardCard -> canPlay -> canActivate.
* Revealed missing;
* Library missing;
* Other ok.